Quran/SQLiteSearchableAyahTextPersistence.swift

```diff
@@ -158,4 +222,18 @@ private extension String {
         }
         return ranges.reversed()
     }
+
+    func regexRanges(of term: String) -> [Range<String.Index>] {
+        var ranges: [Range<String.Index>] = []
+        var maximum: Range<String.Index>?
+        while true {
```

Comment:
Can we use `NSRegularExpression.matches` from [here](https://developer.apple.com/documentation/foundation/nsregularexpression/1412446-matches) instead? We will remove the need for the `while true`.